src/main/java/com/faforever/client/map/MapService.java

```diff
@@ -303,7 +306,12 @@ public MapBean readMap(Path mapFolder) throws MapLoadException {
   @Cacheable(value = CacheNames.MAP_PREVIEW, unless = "#result == null")
   public Image loadPreview(String mapName, PreviewSize previewSize) {
```

Comment:
Just that you know I think having the random icon was a feature and was not done because guys that wrote it did not know how to fetch the preview...

But since you are the maintainer of map gen I leave it to u, just think about it that maybe having the mystery about how the map looks until you join might be interesting.

Reply:
yeah I was just considering whether it should show the preview since we also added the options but am honestly not sure about this yet just implemented to see. This also will only get it for the host before they create the game